Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseCollection.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Collections
+{
+    public abstract class BaseCollection : MonoBehaviour
+    {
+        #region public members
+        /// <summary>
+        /// Action called when collection is updated
+        /// </summary>
+        public Action<BaseCollection> OnCollectionUpdated;
+
+        /// <summary>
+        /// List of objects with generated data on the object.
+        /// </summary>
+        [HideInInspector]
+        public List<CollectionNode> NodeList = new List<CollectionNode>();
+
+        /// <summary>
+        /// Whether to treat inactive transforms as 'invisible'
+        /// </summary>
+        public bool IgnoreInactiveTransforms = true;
```

Comment:
fields should not be public. Please update them to be private and serialized. If a field needs to be accessed outside of the scope of the class, please add a property accessor.